ios/Gekidou/Sources/Gekidou/Storage/Database+Posts.swift

```diff
@@ -106,71 +106,47 @@ struct ThreadSetters {
-        let deleteAtCol = Expression<Int64>("delete_at")
-        var postQuery = postTable
-            .select(createAtCol)
-            .where(channelIdCol == channelId && deleteAtCol == 0)
-        
-        if let earliest = earliest, let latest = latest {
-            postQuery = postQuery.filter(earliest...latest ~= createAtCol)
-        }
-        postQuery = postQuery.order(createAtCol.desc).limit(1)
-        
-        if let result = try db.pluck(postQuery) {
-            return try result.get(createAtCol)
+        let idCol = Expression<String>("id")
+        let lastFetchedAtColAsDouble = Expression<Double?>("last_fetched_at")
+        let lastFetchedAtColAsInt64 = Expression<Int64?>("last_fetched_at")
+        let query = myChannelTable.where(idCol == channelId)
+        
+        if let result = try db.pluck(query) {
+            let lastFetchAtInt64 = result[lastFetchedAtColAsInt64]
+            if lastFetchAtInt64 != nil {
+                return lastFetchAtInt64
+            }
+            if let last = result[lastFetchedAtColAsDouble] {
+                return Int64(last)
+            }
         }
         
         return nil
     }
     
-    private func queryPostsInChannelEarliestAndLatest(_ serverUrl: String, _ channelId: String) throws -> (Int64, Int64) {
-        let db = try getDatabaseForServer(serverUrl)
-        let earliest = Expression<Int64>("earliest")
-        let latest = Expression<Int64>("latest")
-        let id = Expression<String>("channel_id")
-        let query = postsInChannelTable
-            .select(earliest, latest)
-            .where(id == channelId)
-            .order(latest.desc)
-            .limit(1)
+    private func updateMyChannelLastFetchedAt(_ db: Connection, _ channelId: String, _ latest: Int64) throws {
+        let idCol = Expression<String>("id")
+        let lastFetchedAtCol = Expression<Int64>("last_fetched_at")
+        let statusCol = Expression<String>("_status")
         
-
-        for result in try db.prepare(query) {
-            return (try result.get(earliest),
-                    try result.get(latest))
-        }
+        let query = myChannelTable
+            .where(idCol == channelId)
+            .update(lastFetchedAtCol <- latest, statusCol <- "updated")
         
-        return (0, 0)
+        try db.run(query)
     }
-
+    
     public func handlePostData(_ db: Connection, _ postData: PostData, _ channelId: String, _ usedSince: Bool = false, _ receivingThreads: Bool = false) throws {
         let sortedChainedPosts = chainAndSortPosts(postData)
         try insertOrUpdatePosts(db, sortedChainedPosts, channelId)
         let earliest = sortedChainedPosts.first!.create_at
         let latest = sortedChainedPosts.last!.create_at
         if (!receivingThreads) {
             try handlePostsInChannel(db, channelId, earliest, latest, usedSince)
+            
+            let lastFetchedAt = postData.posts.map({max($0.create_at, $0.update_at, $0.delete_at)}).max()
```

Comment:
Is handlePostData only executed after a `fetchChannelsSince`? If not, we may want to reconsider this, because we may be losing messages.

Reply:
This is always called after fetching posts for the channel.

So if the channel has more than 60 posts it uses since, if not it uses fetch posts, same logic as in js